src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -116,6 +117,27 @@ public void TestEmptyPush()
             RunReplication(config, 0, 0);
         }
 
+        [Fact]
+        public void TestPushDocWithFilter()
+        {
+            using (var doc1 = new MutableDocument("doc1"))
+            using (var doc2 = new MutableDocument("doc2")) {
+                doc1.SetString("name", "donotpass");
+                Db.Save(doc1);
+
+                doc2.SetString("name", "pass");
+                Db.Save(doc2);
+            }
+
+            var config = CreateConfig(true, false, false);
+            config.PushFilter = _replicator__filterCallback;
+            RunReplication(config, 0, 0);
+            _isFilteredCallback.Should().BeTrue();
+            _otherDB.GetDocument("doc1").Should().BeNull("because doc1 is filtered out in the callback");
+            _otherDB.GetDocument("doc2").Count().Should().Be(1, "because doc2 is filtered in in the callback");
```

Comment:
This logic is slightly hard to follow.  For these purposes it is enough if the document simply is not null (other tests will test that the push operation actually transmits data correctly)